### PR TITLE
Add VSConfig file

### LINKED ISSUE #6581: Use a .vsconfig file
See here: https://devblogs.microsoft.com/setup/configure-visual-studio-across-your-organization-with-vsconfig/

This way folks don't need to ensure they have the correct workloads installed for VS tooling work; they can just open the solution and get prompted to install the correct components.

### PR DESCRIPTION
**Pull request by cartermp**
Fixes #6581

This should allow someone to open VisualFSharp.sln in VS without knowing exactly what to install and be prompted by VS to install the correct components.

See this blog post for more info: https://devblogs.microsoft.com/setup/configure-visual-studio-across-your-organization-with-vsconfig/

### CODE REVIEW

**Comment by saul** [comment]
This is a great step for first time contributors. Is this as minimal as possible? I see e.g. TypeScript and Docker in there, chose (and any others) be removed if they’re not necessary?

**Comment by cartermp** [comment]
Mea culpa, included some other stuff I installed.

File: .vsconfig [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "Microsoft.VisualStudio.Component.Roslyn.LanguageServices",
+    "Microsoft.VisualStudio.Component.FSharp",
+    "Microsoft.Net.Core.Component.SDK.2.1",
+    "Microsoft.NetCore.ComponentGroup.DevelopmentTools.2.1",
+    "Microsoft.Net.Component.4.7.2.SDK",
+    "Microsoft.Net.Component.4.7.2.TargetingPack",
+    "Microsoft.Net.ComponentGroup.DevelopmentPrerequisites",
+    "Microsoft.Component.MSBuild",
+    "Microsoft.VisualStudio.Component.TextTemplating",
+    "Microsoft.VisualStudio.Component.SQL.CLR",
```
Comment (cartermp) [inline .vsconfig]: Note - yes, this is technically required

File: .vsconfig [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "Microsoft.VisualStudio.Component.Roslyn.Compiler",
+    "Microsoft.VisualStudio.Component.Roslyn.LanguageServices",
+    "Microsoft.VisualStudio.Component.FSharp",
+    "Microsoft.Net.Core.Component.SDK.2.1",
+    "Microsoft.NetCore.ComponentGroup.DevelopmentTools.2.1",
+    "Microsoft.Net.Component.4.7.2.SDK",
+    "Microsoft.Net.Component.4.7.2.TargetingPack",
+    "Microsoft.Net.ComponentGroup.DevelopmentPrerequisites",
+    "Microsoft.Component.MSBuild",
+    "Microsoft.VisualStudio.Component.TextTemplating",
```
Comment (cartermp) [inline .vsconfig]: Yes, this is technically required - legacy .NET projects take a dependency on this for the property pages

**Comment by cartermp** [comment]
```
Command failed to execute with exit code 1: C:\Users\VssAdministrator\.nuget\packages\StrawberryPerl64\5.22.2.1\Tools\perl\bin\perl.exe "D:\a\1\s\tests\fsharpqa\testenv\bin\runall.pl" -resultsroot "D:\a\1\s\artifacts\TestResults\Release" -results test-net40-fsharpqa-results.log -log test-net40-fsharpqa-errors.log -fail test-net40-fsharpqa-errors -cleanup:no -procs:2
System.Management.Automation.RuntimeException: Command failed to execute with exit code 1: C:\Users\VssAdministrator\.nuget\packages\StrawberryPerl64\5.22.2.1\Tools\perl\bin\perl.exe "D:\a\1\s\tests\fsharpqa\testenv\bin\runall.pl" -resultsroot "D:\a\1\s\artifacts\TestResults\Release" -results test-net40-fsharpqa-results.log -log test-net40-fsharpqa-errors.log -fail test-net40-fsharpqa-errors -cleanup:no -procs:2
at Exec-CommandCore, D:\a\1\s\eng\build-utils.ps1: line 50
at Exec-Console, D:\a\1\s\eng\build-utils.ps1: line 119
at <ScriptBlock>, D:\a\1\s\eng\Build.ps1: line 279
```

image

cc @brettfo @KevinRansom

**Comment by cartermp** [comment]
Yay flaky CI. Let's get this in before the machines get angry at me!

**Pull request merged**